doc/nrf/release_notes.rst

```diff
@@ -21,6 +21,7 @@ This page is included only in the latest documentation, because it might contain
    :caption: Subpages:
 
    releases/release-notes-changelog
+   migration/migration_guide_1.x_to_2.x
```

Comment:
I like this placement, as we avoid overcomplicating the structure and we clearly indicate when the migration is supposed to take place.